pyquil/quilatom.py

```diff
@@ -271,7 +308,44 @@ def __hash__(self) -> int:
         return hash(id(self))
 
 
-ParameterDesignator = Union["Expression", "MemoryReference", np.int_, int, float, complex]
+ParameterDesignator = Union["Expression", "MemoryReference", quil_rs.Expression, Number]
```

Comment:
Any reason to not replace `np.int_, int, float, complex` with `Number` here? They all return `True` for `isinstance(x, Number)`. Similarly, I don't think anything we wouldn't want to pass the instance check would get through.

Reply:
checks out to me 👍